pyscriptjs/tests/integration/support.py

```diff
@@ -105,47 +105,73 @@ def init_page(self, page):
         page.set_default_timeout(60000)
 
         self.console = ConsoleMessageCollection(self.logger)
-        self._page_errors = []
-        page.on("console", self.console.add_message)
+        self._js_errors = []
+        page.on("console", self._on_console)
         page.on("pageerror", self._on_pageerror)
 
     def teardown_method(self):
-        # we call check_errors on teardown: this means that if there are still
+        # we call check_js_errors on teardown: this means that if there are still
         # non-cleared errors, the test will fail. If you expect errors in your
         # page and they should not cause the test to fail, you should call
-        # self.check_errors() in the test itself.
-        self.check_errors()
+        # self.check_js_errors() in the test itself.
+        self.check_js_errors()
+
+    def _on_console(self, msg):
+        self.console.add_message(msg.type, msg.text)
 
     def _on_pageerror(self, error):
-        self.logger.log("JS exception", error.stack, color="red")
-        self._page_errors.append(error)
+        self.console.add_message("js_error", error.stack)
+        self._js_errors.append(error)
 
-    def check_errors(self):
+    def check_js_errors(self, *expected_messages):
         """
         Check whether JS errors were reported.
 
-        If it finds a single JS error, raise JsError.
-        If it finds multiple JS errors, raise JsMultipleErrors.
+        expected_messages is a list of strings of errors that you expect they
+        were raised in the page.  They are checked using a simple 'in' check,
+        equivalent to this:
+            if expected_message in actual_error_message:
+                ...
+
+        If an error was expected but not found, it raises
+        DidNotRaiseJsError().
+
+        If there are MORE errors other than the expected ones, it raises an
+        exception:
+
+          - if it finds a single JS error, raise JsError
+          - if it finds multiple JS errors, raise JsMultipleErrors
 
         Upon return, all the errors are cleared, so a subsequent call to
-        check_errors will not raise, unless NEW JS errors have been reported
+        check_js_errors will not raise, unless NEW JS errors have been reported
         in the meantime.
         """
-        exc = None
-        if len(self._page_errors) == 1:
-            # if there is a single error, wrap it
-            exc = JsError(self._page_errors[0])
-        elif len(self._page_errors) >= 2:
-            exc = JsMultipleErrors(self._page_errors)
-        self._page_errors = []
-        if exc:
-            raise exc
-
-    def clear_errors(self):
+        expected_messages = list(expected_messages)
+        js_errors = self._js_errors[:]
```

Comment:
why not just `js_errors = self._js_errors`?

Reply:
because we modify it in-place later (`js_errors[i] = None`).
I think that in this precise case it doesn't change much because we reset `self._js_errors` anyway at the end, but it would make things harder to debug in case there is a bug inside `check_js_errors` itself.
Basically, it's just a good practice.